File: app/lib/utils/format.dart
/// Format angka & tanggal Bahasa Indonesia ala Play Store.
class Fmt {
  /// 999 -> "999", 12.300 -> "12,3 rb", 1.200.000 -> "1,2 jt".
  static String compact(int n) {
    if (n < 1000) return '$n';
    if (n < 1000000) {
      final double v = n / 1000;
      return '${_one(v)} rb';
    }
    final double v = n / 1000000;
    return '${_one(v)} jt';
  }

  static String _one(double v) {
    if (v >= 100) return v.round().toString();
    final String s = v.toStringAsFixed(1).replaceAll('.', ',');
    return s.endsWith(',0') ? s.substring(0, s.length - 2) : s;
  }

  /// "5 mnt lalu", "3 jam lalu", "2 hari lalu", ...
  static String timeAgo(DateTime? dt) {
    if (dt == null) return '-';
    final Duration d = DateTime.now().difference(dt.toLocal());
    if (d.isNegative) return 'baru saja';
    if (d.inSeconds < 60) return 'baru saja';
    if (d.inMinutes < 60) return '${d.inMinutes} mnt lalu';
    if (d.inHours < 24) return '${d.inHours} jam lalu';
    if (d.inDays < 7) return '${d.inDays} hari lalu';
    if (d.inDays < 30) return '${d.inDays ~/ 7} mgg lalu';
    if (d.inDays < 365) return '${d.inDays ~/ 30} bln lalu';
    return '${d.inDays ~/ 365} thn lalu';
  }

  static const List<String> _bulan = [
    '',
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'Mei',
    'Jun',
    'Jul',
    'Agu',
    'Sep',
    'Okt',
    'Nov',
    'Des',
  ];

  /// "12 Sep 2026".
  static String date(DateTime? dt) {
    if (dt == null) return '-';
    final DateTime l = dt.toLocal();
    return '${l.day} ${_bulan[l.month]} ${l.year}';
  }

  /// Rating "4,8" (koma Indonesia).
  static String rating(double r) {
    if (r <= 0) return '-';
    return r.toStringAsFixed(1).replaceAll('.', ',');
  }
}

File: src/lib/format.js
// Format angka & tanggal Bahasa Indonesia ala Play Store.

function one(v) {
  if (v >= 100) return String(Math.round(v));
  const s = v.toFixed(1).replace('.', ',');
  return s.endsWith(',0') ? s.slice(0, -2) : s;
}

export function compact(n) {
  const x = Number(n) || 0;
  if (x < 1000) return String(x);
  if (x < 1000000) return `${one(x / 1000)} rb`;
  return `${one(x / 1000000)} jt`;
}

export function timeAgo(dt) {
  if (!dt) return '-';
  const d = new Date(dt);
  if (Number.isNaN(d.getTime())) return '-';
  const diff = Date.now() - d.getTime();
  if (diff < 0) return 'baru saja';
  const m = Math.floor(diff / 60000);
  if (m < 1) return 'baru saja';
  if (m < 60) return `${m} mnt lalu`;
  const h = Math.floor(m / 60);
  if (h < 24) return `${h} jam lalu`;
  const days = Math.floor(h / 24);
  if (days < 7) return `${days} hari lalu`;
  if (days < 30) return `${Math.floor(days / 7)} mgg lalu`;
  if (days < 365) return `${Math.floor(days / 30)} bln lalu`;
  return `${Math.floor(days / 365)} thn lalu`;
}

const BULAN = [
  '', 'Jan', 'Feb', 'Mar', 'Apr', 'Mei', 'Jun',
  'Jul', 'Agu', 'Sep', 'Okt', 'Nov', 'Des',
];

export function dateFmt(dt) {
  if (!dt) return '-';
  const d = new Date(dt);
  if (Number.isNaN(d.getTime())) return '-';
  return `${d.getDate()} ${BULAN[d.getMonth() + 1]} ${d.getFullYear()}`;
}

export function ratingFmt(r) {
  const x = Number(r) || 0;
  if (x <= 0) return '-';
  return x.toFixed(1).replace('.', ',');
}
